File: perl/HashDemo.pl
#! /bin/perl

use strict;
use NatesHashTable;
use NatesDataObject;

open(my $FILE_HANDLE, 'WORDS_LIST.data') or die("Unable to find words data file");

my @WORDS_LIST = <$FILE_HANDLE>;
chomp(@WORDS_LIST);

close($FILE_HANDLE);

my $wordCount = scalar @WORDS_LIST;

print "$wordCount words loaded, beginning hash...\n";

my $theHashTable = NatesHashTable->new( $wordCount * 2.31 );
my $successCount = 0;

foreach my $word (@WORDS_LIST) {

 	my $obj = NatesDataObject->new($word);
 	if ($theHashTable->insert($obj)) {
 		$successCount++;
 	} else {
 		last;
 	}
}

print "$successCount words hashed\n";

$theHashTable->printMap();

File: java/HashDemo.java
import java.util.Date;

class HashDemo {
	
	public static void main(String[] args) {
		
		System.out.println("Loading Dictionary");
		Dictionary theDictionary = new Dictionary();
		int wordCount = theDictionary.getWordCount();
		int successCount = 0;
		
		System.out.println("Dictionary has " + wordCount + " words");
		
		System.out.println("Beginning hash calculations...");
		
		NatesHashTable theHashTable = new NatesHashTable(wordCount*2.31);
		
		Date startTime = new Date();
		
		for (int i = 0; i < wordCount; i++) {
			if ((successCount % 1000) == 0) {
				System.out.println(successCount + " words hashed");
			}
			if (theHashTable.insert(new NatesDataObject(theDictionary.wordAtIndex(i)))) {
				successCount++;
			} else {
				break;
			};
		}
		
		Date endTime = new Date();
		
		System.out.println("Hashing complete, printing map");
		
		theHashTable.printMap();
		
		System.out.println( successCount + " words hashed in " + (endTime.getTime() - startTime.getTime()) + "ms");
		
	}
	
}